src/include/main/db_main.h

```diff
@@ -77,14 +77,17 @@ class DBMain {
      * @param log_manager argument to the TransactionManager
      */
     TransactionLayer(const common::ManagedPointer<storage::RecordBufferSegmentPool> buffer_segment_pool,
-                     const bool gc_enabled, const common::ManagedPointer<storage::LogManager> log_manager) {
+                     const bool gc_enabled, const bool wal_async_commit_enable,
```

Comment:
What are your thoughts on stashing `wal_async_commit_enable` in the log_manager instead of in `TransactionLayer`?

Reply:
What do you mean by stashing it? The `TransactionManager` needs this dependency so it needs to be injected into its layer. How would `TransactionManager`'s constructor access the boolean value otherwise?

Reply:
Nevermind, I thought the `LogManager` class had getters and setters since I saw `SetNumBuffers` but it doesn't do so for any other private variables.